experiment/nursery/diskcache/cache.go

```diff
@@ -45,10 +49,17 @@ func NewCache(diskRoot string) *Cache {
 	}
 }
 
-func (c *Cache) keyToPath(key string) string {
+// KeyToPath converts a cache entry key to a path on disk
+func (c *Cache) KeyToPath(key string) string {
 	return filepath.Join(c.diskRoot, key)
 }
 
+// PathToKey converts a path on disk to a key, assuming the path is actually
+// under DiskRoot() ...
+func (c *Cache) PathToKey(key string) string {
+	return strings.TrimPrefix(key, c.diskRoot+string(os.PathSeparator))
```

Comment:
You should document that `c.diskRoot` cannot have a trailing slash. Specifying the root dir as the `diskRoot` could be a little strange too since it would need to be specified as "" instead of "/".

Reply:
I'll probably just make the constructor handle this instead, good point though.

Reply:
the constructor now trims os.PathSeperator